### PR TITLE
Adding safeguards for NPEs and certain conditions

### PR DESCRIPTION
**Pull request by jzonthemtn**

### CODE REVIEW

File: opensearch-search-quality-evaluation-plugin/scripts/create-judgments-now.sh [hunk 1]
```diff
@@ -18,4 +18,4 @@ echo "Creating judgments index..."
 #                                              }'
 
 echo "Creating judgments..."
-curl -s -X POST "http://localhost:9200/_plugins/search_quality_eval/judgments?click_model=coec&max_rank=1"
+curl -s -X POST "http://localhost:9200/_plugins/search_quality_eval/judgments?click_model=coec&max_rank=20"
```
Comment (epugh) [inline opensearch-search-quality-evaluation-plugin/scripts/create-judgments-now.sh:21]: so this is to go a lot deeper....?

Comment (jzonthemtn) [reply opensearch-search-quality-evaluation-plugin/scripts/create-judgments-now.sh:21]: Yes, 20 was the original. 1 was for testing.

File: opensearch-search-quality-evaluation-plugin/src/main/java/org/opensearch/eval/judgments/clickmodel/coec/CoecClickModel.java [hunk 2]
```diff
@@ -324,6 +330,7 @@ public Map<Integer, Double> getRankAggregatedClickThrough() throws Exception {
                 final Collection<? extends Terms.Bucket> positionBuckets = positionTerms.getBuckets();
 
                 for(final Terms.Bucket positionBucket : positionBuckets) {
+                    LOGGER.info("Inserting client event from position {} with click count {}", positionBucket.getKey(), (double) positionBucket.getDocCount());
```
Comment (epugh) [inline opensearch-search-quality-evaluation-plugin/src/main/java/org/opensearch/eval/judgments/clickmodel/coec/CoecClickModel.java:333]: will this be crazy verbose?

File: opensearch-search-quality-evaluation-plugin/src/main/java/org/opensearch/eval/judgments/clickmodel/coec/CoecClickModel.java [hunk 3]
```diff
@@ -336,16 +343,28 @@ public Map<Integer, Double> getRankAggregatedClickThrough() throws Exception {
...
         }
 
         for(final Integer x : clickCounts.keySet()) {
-            //System.out.println("Position = " + x + ", Click Count = " + clickCounts.get(x) + ", Event Count = " + viewCounts.get(x));
-            rankAggregatedClickThrough.put(x, clickCounts.get(x) / viewCounts.get(x));
+
+            if(!(impressionCounts.get(x) == null)) {
+
+                // Calculate the CTR by dividing the number of clicks by the number of impressions.
+                LOGGER.info("Position = {}, Click Count = {}, Event Count = {}", x, clickCounts.get(x), impressionCounts.get(x));
```
Comment (epugh) [inline opensearch-search-quality-evaluation-plugin/src/main/java/org/opensearch/eval/judgments/clickmodel/coec/CoecClickModel.java:359]: just wondering if these .info are really .debug?

**Review by epugh — APPROVED** [review]
LGTM.   A couple of questions, but no changes needed.

File: opensearch-search-quality-evaluation-plugin/src/main/java/org/opensearch/eval/judgments/clickmodel/coec/CoecClickModel.java (hunk 3 above)
Comment (jzonthemtn) [reply opensearch-search-quality-evaluation-plugin/src/main/java/org/opensearch/eval/judgments/clickmodel/coec/CoecClickModel.java:359]: Yes, they are. Probably could use a good going-over to make sure level are appropriate.

**Pull request merged**